### PR TITLE
Ubuntu pre-requirements of CRI-O

### PR DESCRIPTION
**Pull request by PauloFerreira25**
Please accept my pull request.

In a minimal installation of ubuntu 22.04, you need to include these 2 packages as pre-requirements of CRI-O.

### CODE REVIEW

**Comment by hswong3i** [comment]
Thanks, but no thanks. This PR will not be accepted:
* Please check https://github.com/alvistack/ansible-role-cri_o/tree/master/vars, in case PR for additional packages required, we should also ensure ALL of our supported OS being patched accordingly, but not just single OS
* Please check https://github.com/alvistack/ansible-role-cri_o/blob/master/molecule/default/converge.yml about which Ansible Roles required for running our molecule test cases, and so how CRI-O could works for real installation. You need much more than just adding crun...
* Please check https://github.com/alvistack/ansible-role-cri_o/blob/master/.gitlab-ci.yml, in case you hope to setup your dev box for running molecule test cases for each OS as our GitLab Runner custom executor. They are target for Ubuntu 22.04, and most of them could just copy and paste and run, with very minimal environment variable customization

**Comment by PauloFerreira25** [comment]
Hello, please let me understand if we think the same thing.

When using the role:
```
   roles:
     - role: alvistack.cri_o
```

I believe there is an expectation that CRI-O will be installed, right?

If you create an installation of ubuntu 22.04, and run this role.

CRI-O does not start due to lack of dependency.

How should I proceed?

**Comment by hswong3i** [comment]
> CRI-O does not start due to lack of dependency.

If running test case with molecule, it will download required dependency for you (see https://github.com/alvistack/ansible-role-cri_o/blob/master/.gitlab-ci.yml#L79), as defined in https://github.com/alvistack/ansible-role-cri_o/blob/master/ansible-galaxy-requirements.yml. Also see https://molecule.readthedocs.io/en/latest/configuration.html#ansible-galaxy

If you are running Ansible manually, check https://galaxy.ansible.com/docs/using/installing.html#installing-multiple-roles-from-a-file 

If you are running for production installation, don't use this Ansible Role (or it dependency) directly, check https://github.com/alvistack/ansible-collection-kubernetes for complete solution (which also now CNCF certified, https://landscape.cncf.io/?selected=alvi-stack-ansible-collection-for-kubernetes). Also see https://github.com/alvistack/ansible-collection-kubernetes/blob/master/playbooks/30-cri_o-install.yml

**Comment by PauloFerreira25** [comment]
I think I got it, your idea of use would be:

roles:
- role: alvistack.containers_common
- role: alvistack.crun
- role: alvistack.cri_o

**Comment by hswong3i** [comment]
> I think I got it, your idea of use would be:

My idea is: If you are running for production installation, don't use this Ansible Role (or it dependency) directly, check https://github.com/alvistack/ansible-collection-kubernetes for complete solution (which also now CNCF certified, https://landscape.cncf.io/?selected=alvi-stack-ansible-collection-for-kubernetes).

**Comment by PauloFerreira25** [comment]
But I'm not running kubernetes.

I'm running minikube in raspberry :)

**Comment by hswong3i** [comment]
> I'm running minikube in raspberry :)

Here my roles are all x86_64 based, depends on my OBS repo deb/rpm, see https://build.opensuse.org/project/show/home:alvistack.

If you are use armv6l or armv7l, it is currently not supported :-(

**Comment by PauloFerreira25** [comment]
Oops, now that I see it, you're packing the projects :(

Why aren't you using the official repositories?

deb https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/
deb http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable:/cri-o

**Comment by hswong3i** [comment]
> Why aren't you using the official repositories?

First of all: coding is my interest, but my code is for profit making. That's why I could accept ANY dirty workaround for fixing my client's today needs, which upstream official repo should NEVER accept it. I need to work in a "Just patch it, NOW!!" style.

### TL;DR
Historically I tried with Google Kubernetes repo + Kubic CRI-O repo, install their static binary with Ansible Role manually, contribute upstream source for static binary packaging with NixOS (~ 1 year), then finally go for my own OBS repo. 

Lesson learned:
1. Google Kubernetes repo conflict with Kubic CRI-O repo. They have different naming for CNI, you couldn't mix them together.
2. Static binary + Ansible Role manually procedure just redo traditional deb/rpm package manager duty dirtily, with no additional values but too complicated
3. Contribute upstream repo static binary packaging with NixOS also too complicated, especially when armv6l or armv7l newly added (and also that's why I give up with this method)
4. Upstream repo usually only keep latest version of crun/conmon/skopeo/buildah/podman/crio/kubelet/kubeadm/kubectl for limited OS, which I need to keep both N/N-1/N-2 version for my multiple supported OS. That is VERY useful, especially latest initial minor release (e.g. 1.25.0) are ALWAYS buggy. If using official repo I am FORCED to update with no choice nor fallback plan

With my own OBS repo I could:
1. Keep both N/N-1/N-2 version for multiple support OS
2. Just patch and package it now, not waiting for 3+ months for PR get merged (e.g. https://github.com/openshift/openshift-restclient-python/pull/426 or https://github.com/ansible-community/molecule-vagrant/pull/174 or https://github.com/ceph/ceph/pull/47714)
3. Apply different patch for different release (which Ansible Role based single post-install alter style couldn't manage it nicely)
4. Manage package content install/upgrade/delete by deb/rpm package manager nicely
5. Support HTTP_PROXY or HTTPS_PROXY in corporate LAN-to-WAN network, and so also support download caching
6. Greatly simplify my Ansible Role implementation logic
7. I could simply manage both deb/rpm and Ansible Role and Ansible Collection in sync

Managing a OBS repo for ~900 package (they just growth since 1 year ago!!) daily is not a funny duty but boring; BTW these dirty workaround greatly increase the service level that I am now providing to my corporate clients.

Back to my origin: coding is my interest, but my code is for profit making.

P.S. I have no paid corporate clients asking production server with Raspberry PI based cluster, but only with their own in-house bare meter or private cloud or public cloud or hybrid cloud in x86_64. So I have no schedule for adding armv6l or armv7l support.

**Comment by PauloFerreira25** [comment]
Hello,
Thank you very much for your attention.
Please excuse me for bothering you.
Don't worry, I'll recreate the ansible modules by myself, including the support for raspberry.
My thought is that using something that is already done is better than redoing it. But if something isn't ready, I do it.
The ideology of your work is at odds with my need. I don't want to interfere with your work.
Again, thanks for your attention.
